plugins/woocommerce/includes/class-wc-order-item-product.php

```diff
@@ -435,6 +435,43 @@ public function get_tax_status() {
 		return $product ? $product->get_tax_status() : ProductTaxStatus::TAXABLE;
 	}
 
+	/**
+	 * Get formatted meta data for the item.
+	 *
+	 * This overrides the parent method to conditionally remove backorder
+	 * meta data when the order is marked as completed.
+	 *
+	 * @param string $hideprefix  Meta data prefix, (default: _).
+	 * @param bool   $include_all Include all meta data, this stop skip items with values already in the product name.
+	 * @return array
+	 */
+	public function get_formatted_meta_data( $hideprefix = '_', $include_all = false ) {
+		$formatted_meta = parent::get_formatted_meta_data( $hideprefix, $include_all );
+
+		$order = $this->get_order();
+
+		if ( $order && $order->has_status( 'completed' ) ) {
```

Comment:
HInt: we have `\Automattic\WooCommerce\Enums\OrderStatus` enum class.

Reply:
Fixed, thank you.